src/OrgasmControl.cpp

```diff
@@ -47,6 +47,30 @@ namespace OrgasmControl {
       }
 
       last_value = p_check;
+
+      // detect muscle clenching.  (Can be used for ruined orgasm if tease threshold is set too high)
+      if (p_check >= (clench_pressure_threshold + Config.clench_pressure_sensitivity) ) {
+        clench_pressure_threshold = (p_check - (Config.clench_pressure_sensitivity/2)); // raise clench threshold to pressure - 1/2 sensitivity
+      }
+      if (p_check >= clench_pressure_threshold) {
+        clench_duration += 1;   // Start counting clench time if pressure over threshold
+        if ( clench_duration > Config.clench_duration_threshold) {
+          arousal += 100;     // boost arousal  because clench duration exceeded
+          if ( arousal > 4095 ) { arousal = 4096; } // protect arousal value to not go higher then 4096
+        }
+        if ( clench_duration >= (Config.clench_duration_threshold*2) ) { // desensitize clench threshold when clench too long. this is to stop arousal from going up
+          clench_pressure_threshold += 400;
+          clench_duration = 0;
+        }
+      } else {                     // when not clenching lower clench time and decay clench threshold
+        clench_duration -= 5;
+        if ( clench_duration <=0 ) {
+          clench_duration = 0;
+          if ( (p_check + (Config.clench_pressure_sensitivity/2)) < clench_pressure_threshold ){  // clench pressure threshold value decays over time to a min of pressure + 1/2 sensitivity
+            clench_pressure_threshold -= 1;
+          }
+        }
+      } // end of clenching detection
```

Comment:
Can you wrap this entire functionality in a new config param, `bool Config.use_clench_detection` or disable if one of the values is 0. By default, software updates shouldn't change device behavior, but allow an opt-in new feature.